api/service/prediction_job_service.go

```diff
@@ -173,6 +173,11 @@ func (p *predictionJobService) StopPredictionJob(ctx context.Context, env *model
 		return nil, err
 	}
 
+	if job.Status.IsTerminal() && job.Status != models.JobTerminated {
```

Comment:
because if we let the ctl.Stop handle some of the terminal status (failed, failed_submission) which doesnt have any spark application, this leads to error when we delete the job

Reply:
Hmm where this this change go? Why was this removed? 🤔

Reply:
Update: i kinda changed the implementation, since the runner is updating for every job, lets just use ctl.Stop to do the deletion, the implementation is kinda different than turing, so when user are terminating the job, it will immediately get deleted, the runner on merlin is working differently than turing, since the worker only works if the spark application is existing (the worker function is to update job status from the spark application status)

The current implementation will immediately cleanup the spark application + do the deletion from the database

Reply:
That makes sense, thanks for the changes!